recipes/libpq/meson/conanfile.py

```diff
@@ -0,0 +1,378 @@
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/[>=2.2 <3]")
+        if self._settings_build.os == "Windows":
+            self.tool_requires("winflexbison/2.5.24")
+            self.tool_requires("strawberryperl/5.32.1.1")
+        else:
+            self.tool_requires("flex/2.6.4")
+            self.tool_requires("bison/3.8.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        VirtualBuildEnv(self).generate()
+
+        feature = lambda option: "enabled" if option else "disabled"
+        true_false = lambda option: "true" if option else "false"
+
+        tc = MesonToolchain(self)
+        tc.project_options["rpath"] = true_false(not self.options.get_safe("disable_rpath"))
+        tc.project_options["bonjour"] = feature(self.options.get_safe("with_bonjour"))
+        tc.project_options["bsd_auth"] = "disabled"
+        tc.project_options["docs"] = "disabled"
+        tc.project_options["dtrace"] = "disabled"
+        tc.project_options["gssapi"] = feature(self.options.with_gssapi)
+        tc.project_options["icu"] = feature(self.options.get_safe("with_icu"))
+        tc.project_options["ldap"] = feature(self.options.with_ldap)
+        tc.project_options["libxml"] = feature(self.options.with_libxml)
+        tc.project_options["libxslt"] = feature(self.options.with_libxslt)
+        tc.project_options["llvm"] = "disabled"
+        tc.project_options["lz4"] = feature(self.options.get_safe("with_lz4"))
+        tc.project_options["nls"] = feature(self.options.with_nls)
+        tc.project_options["pam"] = feature(self.options.get_safe("with_pam"))
+        tc.project_options["plperl"] = "disabled"
+        tc.project_options["plpython"] = "disabled"
+        tc.project_options["pltcl"] = "disabled"
+        tc.project_options["readline"] = feature(self.options.get_safe("with_readline"))
+        tc.project_options["libedit_preferred"] = true_false(self.options.get_safe("with_readline") == "editline")
+        tc.project_options["selinux"] = feature(self.options.get_safe("with_selinux"))
+        tc.project_options["ssl"] = "openssl" if self.options.with_openssl else "none"
+        tc.project_options["systemd"] = feature(self.options.get_safe("with_systemd"))
+        tc.project_options["uuid"] = "e2fs" if self.options.with_uuid else "none"
+        tc.project_options["zlib"] = feature(self.options.with_zlib)
+        tc.project_options["zstd"] = feature(self.options.with_zstd)
+        tc.generate()
+
+        deps = PkgConfigDeps(self)
+        deps.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
+
+    def _fix_static_library_names(self):
+        # Meson outputs static libraries with a .a extension on Windows,
+        # which is not compatible with Conan and CMake.
```

Comment:
Fyi, newer CMake versions, 3.29 and later, support `.a` file extensions for static libraries on Windows when building with MSVC. See the release notes [here](https://cmake.org/cmake/help/latest/release/3.29.html#other-changes).

It would be nice to see support for using the `.a` suffixes eventually. I think Meson made the right decision to use `.a` instead of `.lib` on Windows for static libraries to avoid confusion with import libraries which use the same `.lib` extension.